hazelcast-sql/src/test/java/com/hazelcast/jet/sql/impl/type/AttributePartitioningStrategyTest.java

```diff
@@ -0,0 +1,208 @@
+        final List<HazelcastInstance> nonOwners = getNonOwners(owner);
+
+        assertEquals(10000, owner.getMap("test").getLocalMapStats().getOwnedEntryCount());
+        assertEquals(0, nonOwners.get(0).getMap("test").getLocalMapStats().getOwnedEntryCount());
+        assertEquals(0, nonOwners.get(1).getMap("test").getLocalMapStats().getOwnedEntryCount());
+    }
+
+    @Test
+    public void testCompactObject() {
+        final IMap<GenericRecord, Long> map = instance().getMap("test");
+
+        for (long i = 0; i < 10000; i++) {
+            map.put(GenericRecordBuilder.compact("testType")
+                    .setInt64("id", i)
+                    .setString("name", "key#" + i)
+                    .setString("org", PARTITION_ATTRIBUTE)
+                    .build(), i);
+        }
+
+        final HazelcastInstance owner = getOwner(PARTITION_KEY);
+        final List<HazelcastInstance> nonOwners = getNonOwners(owner);
+
+        assertEquals(10000, owner.getMap("test").getLocalMapStats().getOwnedEntryCount());
+        assertEquals(0, nonOwners.get(0).getMap("test").getLocalMapStats().getOwnedEntryCount());
+        assertEquals(0, nonOwners.get(1).getMap("test").getLocalMapStats().getOwnedEntryCount());
+    }
+
+    @Test
+    public void testPortableObject() {
+        final IMap<GenericRecord, Long> map = instance().getMap("test");
+
+        for (long i = 0; i < 10000; i++) {
+            map.put(GenericRecordBuilder.portable(PORTABLE_CLASS)
+                    .setInt64("id", i)
+                    .setString("name", "key#" + i)
+                    .setString("org", PARTITION_ATTRIBUTE)
+                    .build(), i);
+        }
+
+        final HazelcastInstance owner = getOwner(PARTITION_KEY);
+        final List<HazelcastInstance> nonOwners = getNonOwners(owner);
+
+        assertEquals(10000, owner.getMap("test").getLocalMapStats().getOwnedEntryCount());
+        assertEquals(0, nonOwners.get(0).getMap("test").getLocalMapStats().getOwnedEntryCount());
+        assertEquals(0, nonOwners.get(1).getMap("test").getLocalMapStats().getOwnedEntryCount());
+    }
+
+    @Test
+    public void testJsonObject() {
+        final String template = "{"
+                + "\"id\": <id>,"
+                + "\"name\": \"<name>\","
+                + "\"org\": \"<org>\""
+                + "}";
+        final IMap<HazelcastJsonValue, Long> map = instance().getMap("test");
+
+        for (long i = 0; i < 10000; i++) {
+            final String key = template
+                    .replace("<id>", String.valueOf(i))
+                    .replace("<>", "key#" + i)
+                    .replace("<org>", PARTITION_ATTRIBUTE);
+            map.put(json(key), i);
+        }
+
+        final HazelcastInstance owner = getOwner(PARTITION_KEY);
+        final List<HazelcastInstance> nonOwners = getNonOwners(owner);
+
+        assertEquals(10000, owner.getMap("test").getLocalMapStats().getOwnedEntryCount());
```

Comment:
this does not check if the key is extracted. We would get the same result if the `AttributePartitionStrategy` returned always `null`. We need tests that will have different partition keys.

Reply:
BTW, `AttributePartitionStrategy` is a good candidate for unit tests

Reply:
you are right. I did not expect the logic to be also in HeapData (and similar in NativeMemoryData).